src/SecondHomeWorkApplication.java

```diff
@@ -0,0 +1,40 @@
+public class SecondHomeWorkApplication {
+    public static void main(String[] args) {
+        System.out.println(booleanExpression(false, false, false, false));
+        System.out.println(booleanExpression(true, true, true, true));
+        System.out.println(booleanExpression(false, false, true, true));
+        System.out.println(booleanExpression(true, true, false, false));
+        System.out.println(booleanExpression(true, false, false, false));
+        System.out.println(booleanExpression(true, true, true, false));
+        System.out.println(booleanExpression(false, true, true, false));
+        System.out.println(booleanExpression(true, false, false, true));
+
+        System.out.println(leapYearCount(1));
+        System.out.println(leapYearCount(4));
+        System.out.println(leapYearCount(100));
+
+        System.out.println(doubleExpression(0.1, 0.2, 0.3));
+        System.out.println(doubleExpression(0.0001, 0.0002, 0.0003));
+
+        System.out.println(flipBit(13, 1));
+        System.out.println(flipBit(13, 2));
+        System.out.println(flipBit(13, 3));
+        System.out.println(flipBit(13, 4));
+    }
+
+    public static boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
+        return ((a ^ b) & (c ^ d)) | ((a ^ c) & (b ^ d));
+    }
+
+    public static int leapYearCount(int year) {
+        return year / 4 - year / 100 + year / 400;
+    }
+
+    public static boolean doubleExpression(double a, double b, double c) {
+        return Math.abs((a + b) - c) <= 0.0001;
```

Comment:
create constant in the class for 0.0001:
privat static final ....
https://mate-academy.github.io/style-guides/java/java.html#s5-naming